dask/delayed.py

```diff
@@ -484,13 +484,20 @@ class Delayed(DaskMethodsMixin, OperatorMethodMixin):
     Equivalent to the output from a single key in a dask graph.
     """
 
-    __slots__ = ("_key", "_dask", "_length")
+    __slots__ = ("_key", "_dask", "_length", "_layer")
 
-    def __init__(self, key, dsk, length=None):
+    def __init__(self, key, dsk, length=None, layer=None):
         self._key = key
         self._dask = dsk
         self._length = length
 
+        # NOTE: Layer is used by `to_delayed` in other collections, but not in normal Delayed use
+        self._layer = layer or key
```

Comment:
```suggestion
        if layer:
            self._layer = layer
        elif isinstance(key, tuple):
            self._layer = key[0]
        else:
            self._layer = key
```

Reply:
I intentionally didn't do this because I'd rather be explicit about passing the layer name when it's needed, and not make any assumptions about key structure in Delayed. The `self._layer not in dsk.layers` check below should catch any misuse. Since `__init__` should only be used internally, not directly by users, this seemed reasonable to me.